client/src/connection/session.ts

```diff
@@ -0,0 +1,20 @@
+// Copyright © 2022-2023, SAS Institute Inc., Cary, NC, USA.  All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+import { LogLine, OnLogFn, RunResult } from ".";
+
+export class BaseSession {
+  _onLogFn: OnLogFn;
+  public set onLogFn(value: (logs: LogLine[]) => void) {
```

Comment:
nit: as `OnLogFn` type already defined, it would be better to use it for all type annotations everywhere.

Reply:
Implemented. See Updated.